### PR TITLE
[RNMobile] File block V - Making Download button use RichText

### PR DESCRIPTION
**Pull request by etoledom**
## Description

This PR fixes an issue on iOS where using RichText on the Download Button makes it lose its width.

**NOTE:** We will also need a fix for Android, since it's also not behaving as intended.

These are the issues on Android:
RT-Android

This is how it works on iOS:
RT-iOS

## How has this been tested?

- Add a File block
- Check that the Download button layout is correct.
- Test change the button's text and removing all text to show the placeholder.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/button/edit.native.js [hunk 1]
```diff
@@ -310,13 +282,6 @@ class ButtonEdit extends Component {
 				? borderRadiusValue + spacing + borderWidth
 				: 0;
 
-		// To achieve proper expanding and shrinking `RichText` on iOS, there is a need to set a `minWidth`
-		// value at least on 1 when `RichText` is focused or when is not focused, but `RichText` value is
-		// different than empty string.
-		const minWidth =
-			isButtonFocused || ( ! isButtonFocused && text && text !== '' )
-				? MIN_WIDTH
-				: placeholderTextWidth;
```
Comment (etoledom) [inline packages/block-library/src/button/edit.native.js:319]: This is not needed anymore for iOS and it's working in the same way as it was before, but on Android this change seems to break the placeholder minimum width.

@jd-alexander - we need to test the `Buttons` block also while working on this.

Comment (jd-alexander) [reply packages/block-library/src/button/edit.native.js:319]: @etoledom just to verify could we keep the previous implementation for Android? I did some checks and so far I haven't found a good way to replicate the behavior that's done in the iOS native code as there's no way to easily measure the placeholder text and set that as the width of the text component. I am going to have another look tomorrow to see if anything comes up.

Comment (etoledom) [reply packages/block-library/src/button/edit.native.js:319]: We could, but the previous behavior on Android still looks like the one in the first screenshot (With "Downloa`\n`d").
If we have a solution only for that one and not for placeholder, I guess we can revert the placeholder changes for Android.

**Comment by jd-alexander** [comment]
@etoledom I spent some time looking into this issue today and I am still in the process of finalizing a fix for the RichText issue on Android. I am just sharing this update here with what I have already done. 

1.  I tried to replicate the solution you have on iOS by measuring the placeholder text but I didn't quite get that working as I wanted to. The process I utilized to do this was as follows. The current `ReactAztecText` component has `InputType.TYPE_TEXT_FLAG_MULTI_LINE` so this is what causes the Download word to wrap in the container. 
I created an EditText similar to that of the `ReactAztecText` with the same text size, hint, text, etc. Similar properties to   `ReactAztecText`expect for the `InputType.TYPE_TEXT_FLAG_MULTI_LINE` flag because that would cause this `EditText` to also wrap. My idea here is that if I could somehow cause this shadow `EditText` to be measured, I could get the width of the placeholder text and set that width as the `minWidth` for the `ReactAztecText` component. However, I wasn't able to get the view drawn appropriately. I probably need to add it to the root layout as an invisible object or something else. I didn't dig too deep here because I wanted to ensure that I am not trying to solve a problem in a complex way before getting feedback. 

2. The next idea I have is to simply utilize the previous placeholder implementation as you shared, and try to measure the placeholder length and set that as the minimum length. I started this but I haven't gotten the desired result as yet as the increased `minWidth` was causing some weird behavior with the button component. So I am going to keep investigating this as the preferred solution.

File: packages/block-library/src/button/edit.native.js (hunk 1 above)
Comment (jd-alexander) [reply packages/block-library/src/button/edit.native.js:319]: Okay, yes, I am working on a solution involving this which should be measuring the placeholder and using that value as the minimum witdth. Just to clarify this is how the issues looks on Android.

Placeholder | Text 
--------|-------
<kbd></kbd>     |     <kbd></kbd>

**Comment by etoledom** [comment]
Thanks for the effort @jd-alexander !

I can talk only from the iOS perspective, so not sure if any of this is gonna be helpful:
The placeholder size was just part of the problem. The main problem was that the TextField component used for RichText (Aztec) had width 0. This issue was fixed for iOS with this change:

https://github.com/WordPress/gutenberg/blob/a982c8846037051a2596ed16f526e6d16f170072/packages/block-library/src/file/edit.native.js#L372-L373

But this only works for iOS. On Android, the wrapping text I'm not sure if it has something to do with the placeholder, since it also happens when the component has content. On iOS we had to measure the placeholder separately because we add a placeholder manually (it's not part of Aztec).

@hypest might have good insight about how to approach this issue on RNAztec Android or Aztec itself.

**Comment by hypest** [comment]
> @hypest might have good insight about how to approach this issue on RNAztec Android or Aztec itself.

Sorry, nothing specific from my side. One thing I wonder about though is whether this works with the already ported Button block and whether we can re-use anything from there.

**Comment by etoledom** [comment]
>  One thing I wonder about though is whether this works with the already ported Button block and whether we can re-use anything from there.

We can re-implement the same strategy, but it looks pretty much like patch.
To avoid this issue from taking too long on solving, I have done just that here: https://github.com/WordPress/gutenberg/pull/27275

I'm closing this PR in favour of the new one since it already works for both platforms.

**Pull request closed without merging**